code/modules/library/book.dm

```diff
@@ -103,6 +103,13 @@
 /obj/item/book/proc/on_read(mob/user)
 	if(book_data?.content)
 		user << browse("<meta charset=UTF-8><TT><I>Penned by [book_data.author].</I></TT> <BR>" + "[book_data.content]", "window=book[window_size != null ? ";size=[window_size]" : ""]")
+
+		LAZYINITLIST(user.mind.book_titles_read)
+		var/has_not_read_book = isnull(user.mind.book_titles_read[starting_title])
+
+		if(has_not_read_book) // any new books give bonus mood
+			SEND_SIGNAL(user, COMSIG_ADD_MOOD_EVENT, "book_nerd", /datum/mood_event/book_nerd)
+			user.mind.book_titles_read[starting_title] = TRUE
```

Comment:
Sanity check that `user.mind` exists before this
(I highly doubt that a mindless mob would ever read a book but you never know)

Reply:
Applied with https://github.com/tgstation/tgstation/pull/66371/commits/3ab48dd730fa52c79f0edbe11cabe544db367c61